examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,692 @@
+    app works.
+-   **The TV platform app**: The TV platform app is generally implemented by the
+    TV manufacturer and provides common capabilities around media playback on
+    the TV like the Basic Video Player. The
+    [example Matter tv-app](https://github.com/project-chip/connectedhomeip/tree/master/examples/tv-app)
+    for Android builds on top of the Matter SDK to demonstrate how a TV platform
+    app works.
+
+This document describes how enable your Android and iOS apps to act as a Matter
+"Casting Client". This documentation is also designed to work with the example
+[example Matter tv-casting-app](https://github.com/project-chip/connectedhomeip/tree/master/examples/tv-casting-app)
+samples so you can see the experience end to end.
+
+## Introduction
+
+A Casting Client (e.g. a mobile phone app) is expected to be a Matter
+Commissionable Node and a `CastingPlayer` (i.e. a TV) is expected to be a Matter
+Commissioner. In the context of the
+[Matter Video Player architecture](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc),
+a `CastingPlayer` would map to
+[Casting "Video" Player](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc#1-introduction).
+The `CastingPlayer` is expected to be hosting one or more `Endpoints` (similar
+to
+[Content Apps](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc#1-introduction)
+in the Matter Video Player architecture) that support one or more Matter Media
+`Clusters`.
+
+The steps to start a casting session are:
+
+1. [Initialize](#initialize-the-casting-client) the `CastingClient` using the
+   Matter SDK.
+1. [Discover](#discover-casting-players) `CastingPlayer` devices using Matter
+   Commissioner discovery.
+1. [Connect](#connect-to-a-casting-player) to the `CastingPlayer` to discover
+   available endpoints. By connecting, the 'CastingClient' will send a
+   User Directed Commissioning (UDC) request to the 'CastingPlayer' device in
+   order to make a Matter commissioning request. The
+   'CastingPlayer' will then obtain the appropriate user consent to allow a
+   connection from this 'CastingClient' and obtain the setup code needed to
+   commission the 'CastingClient'. The setup code will typically come from a
+   corresponding TV content app or be input by the user.
+1. [Select](#select-an-endpoint-on-the-casting-player) an available `Endpoint`
+   hosted by the `CastingPlayer`.
+
+Next, you're ready to:
+
+1. [Issue commands](#issuing-commands) to the `Endpoint`.
+1. [Read](#read-operations) endpoint attributes like playback state.
+1. [Subscribe](#subscriptions) to playback events.
+
+## Build and Setup
+
+The Casting Client is expected to consume the Matter TV Casting library built
+for its respective platform to consume any of the APIs described in this
```

Comment:
...for its respective platform which implements the APIs described in this document.

Reply:
Updated in project-chip#29444